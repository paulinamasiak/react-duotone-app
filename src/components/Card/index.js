import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ imageSrc }) => {
  return (
    <div className="card">
      <picture className="thumbnail">
        <img src={imageSrc} alt="" />
      </picture>
    </div>
  );
};

Card.propTypes = {
  imageSrc: PropTypes.string,
};

Card.defaultProps = {
  imageSrc: '',
};

export default Card;
