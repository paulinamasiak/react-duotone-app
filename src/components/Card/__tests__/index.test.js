import React from 'react';
import { shallow } from 'enzyme';
import Card from '../index';

describe('Card component', () => {
  it('should render without crashing', () => {
    // when
    const wrapper = shallow(<Card />);

    // then
    expect(wrapper.exists()).toBe(true);
  });

  it('should render correctly with default props', () => {
    // when
    const wrapper = shallow(<Card />);

    // then
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it('should pass props.imageSrc to image src prop', () => {
    // given & when
    const imageSrc = '/static/media/photo.jpg';
    const wrapper = shallow(<Card imageSrc={imageSrc} />);

    // then
    expect(wrapper.find('img').prop('src')).toBe('/static/media/photo.jpg');
  });
});
