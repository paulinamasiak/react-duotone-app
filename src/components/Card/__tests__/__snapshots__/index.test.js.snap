// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render correctly with default props 1`] = `
Array [
  <div
    className="card"
>
    <picture
        className="thumbnail"
    >
        <img
            alt=""
            src=""
        />
    </picture>
</div>,
]
`;
